### ISSUE TITLE
shebang lines are lost in exactprint

### ISSUE THREAD

**Issue by kokobd** [haskell/haskell-language-server#2724]
In hls, the exactprint doesn't preserve lines starting with `#!`. I've written a failing test case for this in #2725.

### Your environment

Which OS do you use:
> Ubuntu 20.04

Which LSP client (editor/plugin) do you use:
> VSCode

Describe your project (alternative: link to the project):
> hls itself

Compiler Version:
> Currently checked on ghc 8.10.7

### Steps to reproduce

Given this code:
```haskell
import
#! nix-shell --pure -i runghc -p "haskellPackages.ghcWithPackages (hp: with hp; [ turtle ])"
    Data.Monoid ( Monoid )

class Semigroup a => SomeData a
instance SomeData All
```
We can place the cursor on All and select code action "Add All to the import list of Data.Monoid", and see how the file content changes.

### Expected behaviour

This is the correct code after extending the import, without losing the shebang line.
```haskell
import
#! nix-shell --pure -i runghc -p "haskellPackages.ghcWithPackages (hp: with hp; [ turtle ])"
    Data.Monoid ( Monoid, All )

class Semigroup a => SomeData a
instance SomeData All
```

### Actual behaviour
This is what we are getting now.
```haskell
import

    Data.Monoid ( Monoid, All )

class Semigroup a => SomeData a
instance SomeData All
```

**Comment by eddiemundo** [comment]
I think it's only really legit to have the shebangs as the first line of the file, but maybe preserving everything even stuff that makes no sense is worth.

Not sure if it's related to what you're trying to do but I wrote a parser that finds the last pragma location before the first declaration or module statement that uses the ghc lexer. I think with ghc 9.2 annotation stuff some of it can be deleted but not sure.

https://github.com/haskell/haskell-language-server/blob/master/ghcide/src/Development/IDE/Spans/Pragmas.hs

**Comment by kokobd** [comment]
I found this when working on #2694, trying to exact print the whole module. We have existing test cases for shebang lines in the middle of a source file. That's why I found this issue.

`parsePreDecl` looks better than `ghc-exactprint`'s naive parser, but it may require some modifications to be applicable in this use case.

**Comment by michaelpj** [comment]
@alanz do you know if this is still the case or if it ever got fixed?

**Comment by alanz** [comment]
The changelog says it was fixed in ghc-exactprint 0.5.6, courtesy of @lspitzner
